### PR TITLE
fix memory leaks in arangosh connections

### PR DESCRIPTION
**Pull request by jsteemann**

### CODE REVIEW

File: 3rdParty/fuerte/src/HttpConnection.cpp [hunk 1]
```diff
@@ -508,12 +504,16 @@ void HttpConnection<ST>::setTimeout(std::chrono::milliseconds millis) {
     return; // do
   }
   assert(millis.count() > 0);
-  auto self = shared_from_this();
+  // use a weak-ptr to break cycles
+  auto self = weak_from_this();
```
Comment (graetzer) [inline 3rdParty/fuerte/src/HttpConnection.cpp]: There should be the same logic in VSTconnection

**Comment by graetzer** [comment]
Also I think I do not quite get the logic error here, shouldn’t the cyclic dependency be broken every time we reset the timer

**Comment by jsteemann** [comment]
http://jenkins01.arangodb.biz:8080/view/PR/job/arangodb-matrix-pr/618/

**Comment by jsteemann** [comment]
http://jenkins01.arangodb.biz:8080/view/PR/job/arangodb-matrix-pr/621/

**Comment by jsteemann** [comment]
http://jenkins01.arangodb.biz:8080/view/PR/job/arangodb-matrix-pr/629/

**Comment by jsteemann** [comment]
http://jenkins01.arangodb.biz:8080/view/PR/job/arangodb-matrix-pr/635/

**Comment by jsteemann** [comment]
http://jenkins01.arangodb.biz:8080/view/PR/job/arangodb-matrix-pr/640/

**Comment by jsteemann** [comment]
http://jenkins01.arangodb.biz:8080/view/PR/job/arangodb-matrix-pr/641/

**Pull request merged**